### PR TITLE
fix(formatter): respect `IndentWidth` in `IndentStyle::Tab`

### LINKED ISSUE #81: 🐛 `IndentWidth` does not work with `IndentStyle::Tab`
### Environment information

```block
main branch
```

### What happened?

Check the playground.
- [playground 1](https://biomejs.dev/playground?indentWidth=4&code=ZgB1AG4AYwB0AGkAbwBuACAAZgBvAG8AKAApACAAewAKACAAIAAgACAAcgBlAHQAdQByAG4AIABbAAoAIAAgACAAIAAgACAAIAAgACIAMQAyADMANAA1ADYANwA4ADkAMAAiACwACgAgACAAIAAgACAAIAAgACAAIgAxADIAMwA0ADUANgA3ADgAOQAwACIALAAKACAAIAAgACAAIAAgACAAIAAiADEAMgAzADQANQA2ADcAOAA5ADAAIgAsAAoAIAAgACAAIAAgACAAIAAgACIAMQAyADMANAA1ADYANwA4ADkAMAAiACwACgAgACAAIAAgACAAIAAgACAAIgAxADIAMwA0ADUANgA3ADgAOQAwACIALAAKACAAIAAgACAAXQA7AAoAfQAKAA%3D%3D)
- [playground 2](https://biomejs.dev/playground?indentWidth=8&code=ZgB1AG4AYwB0AGkAbwBuACAAZgBvAG8AKAApACAAewAKAAkAZgB1AG4AYwB0AGkAbwBuACAAYgBhAHIAKAApACAAewAKAAkACQBmAHUAbgBjAHQAaQBvAG4AIABmAG8AbwAoACkAIAB7AAoACQAJAAkAZgB1AG4AYwB0AGkAbwBuACAAYgBhAHIAKAApACAAewAKAAkACQAJAAkAZgB1AG4AYwB0AGkAbwBuACAAZgBvAG8AKAApACAAewAKAAkACQAJAAkACQBmAHUAbgBjAHQAaQBvAG4AIABiAGEAcgAoACkAIAB7AAoACQAJAAkACQAJAAkAZgB1AG4AYwB0AGkAbwBuACAAZgBvAG8AKAApACAAewAKAAkACQAJAAkACQAJAAkAZgB1AG4AYwB0AGkAbwBuACAAYgBhAHIAKAApACAAewAKAAkACQAJAAkACQAJAAkACQByAGUAdAB1AHIAbgAgAFsACgAJAAkACQAJAAkACQAJAAkACQAiAGwAbwBuAGcAIgAsAAoACQAJAAkACQAJAAkACQAJAAkAIgBsAG8AbgBnACIALAAKAAkACQAJAAkACQAJAAkACQAJACIAbABvAG4AZwAiACwACgAJAAkACQAJAAkACQAJAAkACQAiAGwAbwBuAGcAIgAsAAoACQAJAAkACQAJAAkACQAJAAkAIgBsAG8AbgBnACIALAAKAAkACQAJAAkACQAJAAkACQAJACIAbABvAG4AZwAiACwACgAJAAkACQAJAAkACQAJAAkACQAiAGwAbwBuAGcAIgAsAAoACQAJAAkACQAJAAkACQAJAF0AOwAKAAkACQAJAAkACQAJAAkAfQAKAAkACQAJAAkACQAJAH0ACgAJAAkACQAJAAkAfQAKAAkACQAJAAkAfQAKAAkACQAJAH0ACgAJAAkAfQAKAAkAfQAKAH0ACgA%3D)

### Expected result

The result of prettier is the expected behavior.
Although tab is only one character, it should treat indent width as visual width, which will affect the calculation of line width.

Let's paste the biome result into editor.

You will find that the visual width of the code exceeds the limit of 80 characters.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

### PR DESCRIPTION
**Pull request by magic-akari**
## Summary

- fix #81 

## Test Plan

test cases added

### CODE REVIEW

**Comment by magic-akari** [comment]
There are some inconsistent terms, about `indent width` and `indent size`.

I used `IndentWidth` in the implementation.

In the playground, it is called `indent width`. 
But in the cli, it is called `--indent-size`,

This may confuse users, should they be unified?

File: packages/@biomejs/backend-jsonrpc/src/workspace.ts [hunk 1]
```diff
@@ -74,7 +74,7 @@ export interface FilesConfiguration {
 	 */
 	ignoreUnknown?: boolean;
 	/**
-	 * The maximum allowed size for source code files in bytes. Files above this limit will be ignored for performance reason. Defaults to 1 MiB
+	 * The maximum allowed size for source code files in bytes. Files above this limit will be ignored for performance reasons. Defaults to 1 MiB
```
Comment (magic-akari) [inline packages/@biomejs/backend-jsonrpc/src/workspace.ts:77]: I have no idea why this change was requested.

Comment (ematipico) [reply packages/@biomejs/backend-jsonrpc/src/workspace.ts:77]: It's possible that your fork is not up to date with `upstream`

**Review by ematipico — APPROVED** [review]
Looks good to me, this is a great fix!

**Pull request merged**